File: library/src/test/java/me/proxer/library/api/EnumRetrofitConverterFactoryTest.java
package me.proxer.library.api;

import me.proxer.library.entity.info.Entry;
import me.proxer.library.enums.MediaLanguage;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import retrofit2.Converter;
import retrofit2.Retrofit;

import java.io.IOException;
import java.lang.annotation.Annotation;

import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;

/**
 * @author Ruben Gees
 */
class EnumRetrofitConverterFactoryTest {

    private EnumRetrofitConverterFactory factory;
    private Retrofit retrofit;

    @BeforeEach
    void setUp() {
        factory = new EnumRetrofitConverterFactory();
        retrofit = new Retrofit.Builder().baseUrl("https://proxer.me").build();
    }

    @Test
    void testCreate() {
        assertThat(factory.stringConverter(MediaLanguage.class, new Annotation[0], retrofit)).isNotNull();
    }

    @Test
    void testCreateNoEnum() {
        assertThat(factory.stringConverter(Entry.class, new Annotation[0], retrofit)).isNull();
    }

    @SuppressWarnings({"unchecked", "ConstantConditions"})
    @Test
    void testConvert() throws IOException {
        final Converter<Enum<?>, String> converter = (Converter<Enum<?>, String>)
                factory.stringConverter(MediaLanguage.class, new Annotation[0], retrofit);

        assertThat(converter.convert(MediaLanguage.GERMAN)).isEqualTo("de");
    }
}

File: library/src/test/kotlin/me/proxer/library/api/EnumRetrofitConverterFactoryTest.kt
package me.proxer.library.api

import me.proxer.library.entity.info.Entry
import me.proxer.library.enums.MediaLanguage
import org.assertj.core.api.AssertionsForInterfaceTypes.assertThat
import org.junit.jupiter.api.Test
import retrofit2.Retrofit

/**
 * @author Ruben Gees
 */
class EnumRetrofitConverterFactoryTest {

    private val factory = EnumRetrofitConverterFactory()
    private val retrofit = Retrofit.Builder().baseUrl("https://proxer.me").build()

    @Test
    fun testCreate() {
        assertThat(factory.stringConverter(MediaLanguage::class.java, emptyArray(), retrofit)).isNotNull
    }

    @Test
    fun testCreateNoEnum() {
        assertThat(factory.stringConverter(Entry::class.java, emptyArray(), retrofit)).isNull()
    }

    @Test
    fun testConvert() {
        val converter = factory.stringConverter(MediaLanguage::class.java, emptyArray(), retrofit)

        assertThat(converter?.convert(MediaLanguage.GERMAN)).isEqualTo("de")
    }
}
